### PR TITLE
Add DeleteRangeCF for two WriteBatch::Handler subclass

### PR DESCRIPTION
**Pull request by luqun**
Background: We try to use DeleteRange to improve performance for myrocks' truncate table.

During testing, found out there are WriteBatch::Handler subclass don't implement DeleteRangeCF correctly.

This PR is to unblock myself first --- Currently I only need the following two subclasses and don't need range lock no due to the table itself is locked during truncate table

In future, Hope someone can add full support for the following WriteBatch::Handler subclass.

TODO:
1. Range lock support:
- pessimistic_transaction.cc:   class Handler : public WriteBatch::Handler
- write_prepared_txn_db.h: struct SubBatchCounter : public WriteBatch::Handler

2. Transaction support:
- transaction_base.cc:  struct IndexedWriteBatchBuilder : public WriteBatch::Handler 
- write_prepared_txn.cc:  struct RollbackWriteBatchBuilder : public WriteBatch::Handler 
- write_unprepared_txn_db.h: struct KeySetBuilder : public WriteBatch::Handler 
- write_unprepared_txn_db.cc:   struct RollbackWriteBatchBuilder : public WriteBatch::Handler 

Thanks Manuel and Andrew's help.

### CODE REVIEW

**Comment by ajkr** [comment]
Will this make RocksDB appear to support transactional `DeleteRange` while range deletions are actually non-transactional? It feels a bit confusing to me but @maysamyabandeh will be better able to advise.

File: utilities/transactions/write_prepared_txn_db.h [hunk 1]
```diff
@@ -786,6 +786,13 @@ struct SubBatchCounter : public WriteBatch::Handler {
     AddKey(cf, key);
     return Status::OK();
   }
+
+  Status DeleteRangeCF(uint32_t /*column_family_id*/,
+                            const Slice& /*begin_key*/,
+                            const Slice& /*end_key*/) {
+    // TODO: Add range lock support
```
[the remark this answers is not in the document]
Comment (luqun) [reply utilities/transactions/write_prepared_txn_db.h]: Manual, You are right and update the change with your comment

**Comment by maysamyabandeh** [comment]
@luqun It is a good idea to keep PR description concise and to-the-point. Can you rewrite that to explain what this PR is doing in a few sentences?

**Comment by maysamyabandeh** [comment]
Blocker: https://github.com/facebook/rocksdb/issues/4812

**Comment by maysamyabandeh** [comment]
Should be revisited when DeleteRange is properly supported with transactions: #4812

**Pull request closed without merging**